src/controllers/directorDateOfBirth.controller.js

```diff
@@ -0,0 +1,159 @@
+      if (errors && errors.data.details) {
+        pageContext.pageTitle = `${Constants.PAGE_TITLE_ERROR_PREFIX} ${pageContext.pageTitle}`
+      }
+    }
+
+    pageContext.hasDirectors = pageContext.directors.length > 0
+
+    if (request.payload) {
+      for (let i = 0; i < directors.length; i++) {
+        let field = request.payload[`director-dob-day-${i}`]
+        pageContext.directors[i].dob.day = field || ''
+      }
+    }
+    return reply.view('directorDateOfBirth', pageContext)
+  }
+
+  async doPost (request, reply, errors) {
+    const authToken = CookieService.getAuthToken(request)
+    const applicationId = CookieService.getApplicationId(request)
+
+    let account = await Account.getByApplicationId(authToken, applicationId)
+    if (!account) {
+      LoggingService.logError(`Application ${applicationId} does not have an Account`, request)
+      return reply.redirect(Constants.Routes.ERROR.path)
+    }
+
+    const directors = await this._getDirectors(authToken, account.id)
+
+    // Perform manual (non-Joi) validation of dynamic form content
+    errors = await this._validateDynamicFormContent(request, directors)
+
+    if (errors && errors.data.details) {
+      return this.doGet(request, reply, errors)
+    } else {
+      // Save Director dates of birth
+      for (let i = 0; i < directors.length; i++) {
+        const director = directors[i]
+        director.dob.day = request.payload[`director-dob-day-${i}`]
+        await director.save(authToken)
+      }
+
+      return reply.redirect(Constants.Routes.COMPANY_DECLARE_OFFENCES.path)
+    }
+  }
+
+  async _getDirectors (authToken, accountId) {
+    let contacts = await Contact.list(authToken, accountId, Constants.Dynamics.COMPANY_DIRECTOR)
+
+    for (let director of contacts) {
+      if (director.dob && director.dob.month && director.dob.year) {
+        let month = director.dob.month.toString()
+        if (month && month.length === 1) {
+          // Pad with a leading zero if required
+          month = '0' + month
+        }
+        director.dateOfBirthFormatted = moment(`${director.dob.year}-${month}-01`).format('MMMM YYYY')
+      } else {
+        director.dateOfBirthFormatted = 'Unknown date'
+      }
+    }
+    return contacts
+  }
+
+  async _validateDynamicFormContent (request, directors) {
+    let errors
+
+    // If there are are no DOBs entered
+    if (Object.keys(request.payload).length === 0) {
+      errors = {
+        data: {
+          details: [
+            {
+              message: '"director-dobs-not-entered" is required',
+              path: ['director-dobs-not-entered'],
+              type: 'any.required',
+              context: { key: 'director-dobs-not-entered', label: 'director-dobs-not-entered' }
+            }]
+        }
+      }
+    } else {
+      errors = {
+        data: {
+          details: []
+        }
+      }
+
+      // Validate the entered DOB for each director
```

Comment:
You could simplify by defining:

`const directorDobYear = `director-dob-day-${i}`
`

Reply:
Done (I called it directorDobField)